### PR TITLE
Add support for specifying compression level

### PR DESCRIPTION
**Pull request by dmitryvk**
Hi,

I've been using zip-rs to compress somewhat large files on the fly.
Setting compression level from 6 (the default) to 1 makes a big difference in time to compress (about ~3-fold reduction).

In #212 some concerns regarding API compatibility were raised, but I don't see any concerns: field is added to ZipFileData which is not exported from the crate.

I also don't like using u32 to specify compression level (without range checking) - but it seems that flate2, bzip and zstd also use u32.

### CODE REVIEW

File: src/write.rs
Comment (zamazan4ik) [inline src/write.rs:161]: I suggest to leave the information about compression level only in this place and remove duplication in other places. It will reduce complication in future maintenance.

**Review by zamazan4ik — CHANGES_REQUESTED** [review]

**Comment by zamazan4ik** [comment]
Thanks for the PR! I am really glad to see this kind of functionality in the library.

Regarding range check question. Honestly, I also would like to see some kind of range checking. Maybe we can at least provide to the user some kind of helper. But in this case, we will introduce some additional kind of coupling between our library and dependent crates (we will need our range checkers with their supported ranges for each algorithm). I cannot say right now, which way is better.

File: src/write.rs
Comment (dmitryvk) [reply src/write.rs]: Agreed; removed other doc comments

**Comment by dmitryvk** [comment]
I've updated the PR.
Extra doc comments are removed.
I've added range checking for compression level. Values for deflate and bzip2 seem to be standardized, but zstd is somewhat complicated.
It seems that over time compression level range of zstd was increased: zstd crate mentions levels 1-21 and zstd docs state that maximum is 22 and negative values are allowed. Wikipedia states that -7 is the lowest value. Looking at zstd source code it seems that the actual limit is much lower (seems to be -128K though I didn't 100% verify it). Relaxing the constraints is easier so I went with more conservative range of -7 to 22.

**Comment by dmitryvk** [comment]
> But in this case, we will introduce some additional kind of coupling between our library and dependent crates

Yes, this is a coupling, but it's already present. Also, all of the crates provide means to query for allowed compression level values (`flate2::Compression::none/best`, `bzip2::Compression::none/best`, `zstd::compression_level_range`)

**Review by zamazan4ik — APPROVED** [review]

**Comment by zamazan4ik** [comment]
> Yes, this is a coupling, but it's already present. Also, all of the crates provide means to query for allowed compression level values (flate2::Compression::none/best, bzip2::Compression::none/best, zstd::compression_level_range)

Yeah, that is what I wanted to see - using for the range checks ranges directly from our dependencies and does not introduce such ranges in our codebase. 

LGTM.

Since this contribution is quite important from the feature perspective, I suggest wait for some time for the review from @Plecra

**Review by Plecra — APPROVED** [review]
I like it :) Thanks for the changes

**Comment by dmitryvk** [comment]
Thanks, I didn't expect such quick response:)

**Pull request merged**
